xonsh/prompt/vc.py

```diff
@@ -175,19 +164,26 @@ def current_branch():
 
 
 def _git_dirty_working_directory(q, include_untracked):
-    status = None
     denv = builtins.__xonsh__.env.detype()
     try:
-        cmd = ["git", "status", "--porcelain"]
+        # Borrowing this conversation
+        # https://github.com/sindresorhus/pure/issues/115
         if include_untracked:
-            cmd.append("--untracked-files=normal")
+            cmd = [
+                "git",
+                "status",
+                "--porcelain",
+                "--quiet",
+                "--untracked-files=normal",
+            ]
         else:
-            cmd.append("--untracked-files=no")
-        status = subprocess.check_output(cmd, stderr=subprocess.DEVNULL, env=denv)
+            cmd = ["git", "diff", "--no-ext-diff", "--quiet"]
+        child = subprocess.run(cmd, stderr=subprocess.DEVNULL, env=denv)
+        dwd = bool(child.returncode)
```

Comment:
Returncodes don't follow normal boolean rules.  0 is True and non-zero is False

Reply:
HA derp thank you so much!

Reply:
actually i'm sorry, i'm not convinced this is wrong.

https://git-scm.com/docs/git-diff

quiet means it implies --exit-code. Here are the results of --exit-code
- 1 means differences. Therefore, we want dwd to be True
- 0 means no differences. Therefore, we want dwd to be False

Reply:
i chose --quiet because it's supposed to be a bit faster. but is this unintuitive to read? should i leave more comments or handle things differently?

Reply:
More comments would be helpful. In xonsh, returncodes are always interpreted as above, so this would need some clarification. Thanks!